### ISSUE TITLE
500 internal server error when viewing a client without error log

### ISSUE THREAD

**Issue by Plaschke-Culturall** [SolidInvoice/SolidInvoice#467]
Hi!
I just installed Solidinvoice 2.1.1 on Debian 11 with Apache 2.4.54, PHP 7.4.30 and MariaDB 10.5.15
The setup was successful and I can create a new client. But after i click on "Save Client" I get a 500 internal server error.

When I go back to the list of clients, the new client shows up but when I try to view it I also get the 500 error. (Also when I try to create a new invoice after that). Editing this client on the other hand does NOT give me a 500 error.

The weird thing is that I don't see anything in the apache logs (/var/log/apache2/error.log in our case).
I also tried to change
    log_errors = On
in the php.ini and specify a logfile. But still nothing. 
    

Any idea what the problem could be? 
Are there any other locations where I could find logs? Does Solidinvoice itself have logfiles?

**Comment by pierredup** [comment]
Hey @Plaschke-Culturall. Thanks for the report.
In the SolidInvoice directory, there should be logs available in the `var/log` directory (usually `var/log/prod.log`). I'll try to replicate the issue and see if I can spot the problem, but if you can attach the log file, that would help a lot.

**Comment by Plaschke-Culturall** [comment]
Thanks for the quick reply!

The logfile var/log/prod.log doesn't get any new lines when I reproduce the error 500. 
It currently only contains the following lines:
_______________________________________________________________
root@debian11:/var/www/solidinvoice/var/log# tail -100 prod.log
[2022-09-30T15:09:18.166529+02:00] console.CRITICAL: Error thrown while running command "cron:run -e prod -n~". Message: "The "-~" option does not exist." {"exception":"[object] (Symfony\\Component\\Console\\Exception\\RuntimeException(code: 0): The \"-~\" option does not exist. at /var/www/solidinvoice/vendor/symfony/console/Input/ArgvInput.php:118)","command":"cron:run -e prod -n~","message":"The \"-~\" option does not exist."} []
[2022-09-30T15:09:18.175542+02:00] console.DEBUG: Command "cron:run -e prod -n~" exited with code "1" {"command":"cron:run -e prod -n~","code":1} []
_______________________________________________________________

They are from when I first made a mistake with the cronjob (the tilde at the end of the line). But I corrected that afterwards.
The crontab entry now reads: 
`* * * * * php /var/www/solidinvoice/bin/console cron:run -e prod -n`



The syslog then contains lines like
...
Oct  3 11:53:01 debian11 CRON[49752]: (root) CMD (php /var/www/solidinvoice/bin/console cron:run -e prod -n)
Oct  3 11:54:01 debian11 CRON[49763]: (root) CMD (php /var/www/solidinvoice/bin/console cron:run -e prod -n)
Oct  3 11:55:01 debian11 CRON[49773]: (root) CMD (php /var/www/solidinvoice/bin/console cron:run -e prod -n)
...

Which should be ok, right?


I also tried to set 'log_errors' to On in the php.ini and redirected it to /var/www/solidinvoice/php_error.log but this logfile is also empty.

I can't think of any more logfiles where errors could get sent to. Maybe I messed something up during the installation but I just can't find it ...

**Comment by Plaschke-Culturall** [comment]
I did the whole installation again and now I am able to create, view and edit a client.
I can also create a new invoice, but when I try to edit this invoice I again get a 500 internal server error.
The logfile /var/www/solidinvoice/var/log/prod.log now reads:
_____________________________________________________________________________
[2022-10-04T12:42:47.282338+02:00] request.INFO: Matched route "_invoices_edit". {"route":"_invoices_edit","route_parameters":{"_route":"_invoices_edit","_controller":"SolidInvoice\\InvoiceBundle\\Action\\Edit","id":"1"},"request_uri":"http://192.168.0.24/invoices/edit/1","method":"GET"} []
[2022-10-04T12:42:47.283662+02:00] security.DEBUG: Read existing security token from the session. {"key":"_security_frontend","token_class":"Symfony\\Component\\Security\\Core\\Authentication\\Token\\UsernamePasswordToken"} []
[2022-10-04T12:42:47.284381+02:00] security.DEBUG: User was reloaded from a user provider. {"provider":"Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider","username":"plaschke"} []
[2022-10-04T12:42:47.311934+02:00] request.CRITICAL: Uncaught PHP Exception Twig\Error\RuntimeError: "An exception has been thrown during the rendering of a template ("Unable to generate a URL for the named route "" as such route does not exist.")." at /Users/pierre/projects/SolidWorx/SolidInvoice/build/SolidInvoice/src/InvoiceBundle/Resources/views/Default/edit.html.twig line 21 {"exception":"[object] (Twig\\Error\\RuntimeError(code: 0): An exception has been thrown during the rendering of a template (\"Unable to generate a URL for the named route \"\" as such route does not exist.\"). at /Users/pierre/projects/SolidWorx/SolidInvoice/build/SolidInvoice/src/InvoiceBundle/Resources/views/Default/edit.html.twig:21)\n[previous exception] [object] (Symfony\\Component\\Routing\\Exception\\RouteNotFoundException(code: 0): Unable to generate a URL for the named route \"\" as such route does not exist. at /var/www/solidinvoice/vendor/symfony/routing/Generator/CompiledUrlGenerator.php:50)"} []
[2022-10-04T12:42:47.313920+02:00] security.DEBUG: Stored the security token in the session. {"key":"_security_frontend"} []
_____________________________________________________________________________

I enabled the rewrite module of apache (a2enmod rewrite) and my virtual host configuration looks like this:
```
<VirtualHost *:80>
    ServerAdmin webmaster@localhost
    ServerName solidinvoice
    DocumentRoot /var/www/solidinvoice/public

    <Directory /var/www/solidinvoice/public>
        AllowOverride All
        Order allow,deny
        Allow from All
    </Directory>

    ErrorLog /var/log/apache2/error.log
    CustomLog /var/log/apache2/access.log combined
</VirtualHost>
```


Am I missing something?

**Comment by pierredup** [comment]
@Plaschke-Culturall Apologies for the delay, I managed to reproduce the error, and have a fix available in #508. It will be released as part of the 2.1.2 release
